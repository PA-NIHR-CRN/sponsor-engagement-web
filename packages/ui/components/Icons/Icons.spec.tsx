import { render } from '@testing-library/react'
import * as Icons from '.'

describe('NIHR Icon Set', () => {
  test.each(Object.keys(Icons))('Displays the %p correctly', async (key) => {
    const icon = await import(`./${key}`)
    const Component = icon[key]
    expect(render(<Component />).baseElement).toMatchSnapshot()
  })
})
