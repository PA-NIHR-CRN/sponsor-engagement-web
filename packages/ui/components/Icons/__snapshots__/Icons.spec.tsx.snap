// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NIHR Icon Set Displays the "AlertIcon" correctly 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="h-[35px] w-[35px]"
      data-testid="alert-icon"
      fill="currentColor"
      height="35"
      width="35"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#0B0C0C"
        height="35"
        rx="17.5"
        width="35"
      />
      <path
        d="M15.259 6.994h4.438l-.38 13.901h-3.663l-.395-13.9zm-.572 18.486c0-.38.074-.737.22-1.069.147-.342.347-.64.6-.893.264-.254.567-.454.909-.601.342-.147.708-.22 1.099-.22.38 0 .737.073 1.069.22a2.812 2.812 0 011.494 1.494c.147.332.22.689.22 1.07 0 .39-.073.756-.22 1.098-.146.342-.347.64-.6.894a2.82 2.82 0 01-.894.6 2.62 2.62 0 01-1.07.22 2.75 2.75 0 01-1.098-.22 3.002 3.002 0 01-.908-.6 2.822 2.822 0 01-.6-.894 2.759 2.759 0 01-.22-1.099z"
        fill="#fff"
      />
    </svg>
  </div>
</body>
`;

exports[`NIHR Icon Set Displays the "CrossIcon" correctly 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="h-[1em] w-[1em]"
      data-testid="icon-cross"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18 6L6 18M6 6l12 12"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </div>
</body>
`;

exports[`NIHR Icon Set Displays the "HomeIcon" correctly 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="h-[21px] w-[21px]"
      data-testid="home-icon"
      fill="currentColor"
      height="17"
      viewBox="0 0 20 17"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
        stroke="none"
        stroke-width="1"
      >
        <g
          fill-rule="nonzero"
          transform="translate(-22 -88)"
        >
          <g
            transform="translate(20 85)"
          >
            <path
              d="M10 20L10 14 14 14 14 20 19 20 19 12 22 12 12 3 2 12 5 12 5 20z"
            />
          </g>
        </g>
      </g>
    </svg>
  </div>
</body>
`;

exports[`NIHR Icon Set Displays the "SettingsIcon" correctly 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="h-[20px] w-[20px]"
      fill="currentColor"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.896 8.275a.41.41 0 00-.273-.163l-2.383-.365a7.846 7.846 0 00-.534-1.276c.156-.217.39-.523.703-.918a25.3 25.3 0 00.664-.866.501.501 0 00.104-.299.42.42 0 00-.091-.286c-.313-.443-1.029-1.181-2.149-2.214a.502.502 0 00-.325-.13.423.423 0 00-.313.117l-1.848 1.393a8.048 8.048 0 00-1.172-.482L11.914.391a.386.386 0 00-.15-.28.491.491 0 00-.319-.111h-2.89c-.252 0-.408.121-.469.365-.113.433-.239 1.24-.378 2.421a7.45 7.45 0 00-1.185.495L4.727 1.888a.554.554 0 00-.339-.13c-.19 0-.601.31-1.23.93-.63.622-1.057 1.088-1.283 1.4a.548.548 0 00-.117.3c0 .104.043.208.13.312a29.647 29.647 0 011.393 1.797c-.217.4-.386.8-.507 1.198L.352 8.06a.393.393 0 00-.248.17.5.5 0 00-.104.299v2.89c0 .113.035.215.104.306a.41.41 0 00.274.163l2.383.352c.121.425.299.855.533 1.289-.156.217-.39.523-.703.918a26.02 26.02 0 00-.664.865.502.502 0 00-.013.6c.339.468 1.055 1.197 2.149 2.187a.449.449 0 00.325.143.47.47 0 00.326-.117l1.836-1.393c.356.182.746.343 1.172.482l.364 2.395a.387.387 0 00.15.28.49.49 0 00.319.111h2.89c.253 0 .409-.121.47-.365.112-.433.238-1.24.377-2.421a7.36 7.36 0 001.185-.495l1.797 1.406a.636.636 0 00.338.117c.191 0 .6-.308 1.224-.924.625-.617 1.055-1.085 1.29-1.407a.435.435 0 00.116-.299.501.501 0 00-.13-.326 32.755 32.755 0 01-1.393-1.797c.174-.32.343-.716.508-1.184l2.409-.365a.397.397 0 00.26-.17.503.503 0 00.104-.299v-2.89a.49.49 0 00-.104-.306zm-7.539 4.082a3.21 3.21 0 01-2.357.976c-.92 0-1.705-.325-2.357-.976A3.211 3.211 0 016.667 10c0-.92.325-1.706.976-2.357A3.212 3.212 0 0110 6.667c.92 0 1.706.325 2.357.976.651.651.977 1.437.977 2.357a3.21 3.21 0 01-.977 2.357z"
      />
    </svg>
  </div>
</body>
`;

exports[`NIHR Icon Set Displays the "StartIcon" correctly 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="govuk-button__start-icon"
      data-testid="start-icon"
      focusable="false"
      height="19"
      viewBox="0 0 33 40"
      width="17.5"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h13l20 20-20 20H0l20-20z"
        fill="currentColor"
      />
    </svg>
  </div>
</body>
`;
